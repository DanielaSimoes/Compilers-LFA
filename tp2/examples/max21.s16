/* Programa escrito em S16/N2 */

program MAX ()
{
    int[10] in;
    exit();
}

