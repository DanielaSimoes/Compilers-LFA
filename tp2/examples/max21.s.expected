               ; creating a space
               .data
in:            .space 40

               ; exiting program
               .text
L0:
               halt

               ; includes
               halt
printInt:
               ipush 28
printInt1:     dup2
               iushr
               ipush 0xF
               iand
               jsr pNibble
               dup
               ifeq printInt2
               ipush 4
               isub
               goto printInt1
printInt2:     pop
               ret

pNibble:
               ipush 0x30
               iadd
               dup
               ipush 0x39
               isub
               ifle pNibble1
               ipush 0x0a
               iadd
pNibble1:      write
               ret

printStr:      dup
               ipush 0
               baload
               dup
               ifeq printStr_end
               write
               ipush 1
               iadd
               goto printStr

printStr_end:  write
               ret
